src/ds/dllist.h

```diff
@@ -6,16 +6,24 @@
 
 namespace snmalloc
 {
-  template<class T, uintptr_t terminator = 0>
+  template<class T, uintptr_t terminator_ = 0>
   class DLList
   {
   private:
+    // Single point to perform this cast.
+    // Would like this to be a constexpr, but reinterpret cast is not allowed in
+    // constexpr.
+    static inline T* terminator()
+    {
+      return (T*)terminator_;
```

Comment:
Please can we avoid new C-style casts?  I think this wants to be a `static_cast`.